src/browser/remote_subscriptions.ts

```diff
@@ -212,6 +210,101 @@ function unSubscribe(subscription: RemoteSubscription, runtime: PeerRuntime) {
     unSubscriptions.delete(runtimeKey);
 }
 
+/**
+ * Apply all pending remote subscriptions for a given runtime
+ */
+export function applyAllRemoteSubscriptions(runtime: PeerRuntime) {
+    pendingRemoteSubscriptions.forEach(subscription => {
+        if (!subscription.isSystemEvent) {
+            applyRemoteSubscription(subscription, runtime);
+        } else {
+            applySystemSubscription(subscription, runtime);
+        }
+    });
+}
+
+/**
+ * Handles addition of a system subscription on remote runtimes.
+ */
+export function subscribeToAllRuntimes(subscriptionProps: RemoteSubscriptionProps|RemoteSubscription): Promise<() => void> {
+    return new Promise(resolve => {
+        if (systemEventsToIgnore[subscriptionProps.eventName]) {
+            resolve();
+        }
+
+        const clonedProps = Object.create(subscriptionProps);
+        const subscription: RemoteSubscription = Object.assign(clonedProps, {isSystemEvent: true});
+
+        // Generate a subscription ID for pending subscriptions
+        subscription._id = getId();
+
+        // Create an un-subscription map
+        subscription.unSubscriptions = new Map();
+
+        // Subscribe in all connected runtimes
+        if (connectionManager.connections.length) {
+            connectionManager.connections.forEach(runtime => applySystemSubscription(subscription, runtime));
+        }
+
+        // Add the subscription to pending to cover any runtimes launched in the future
+        pendingRemoteSubscriptions.set(subscription._id, subscription);
+
+        // Resolving with a subscription cleanup function
+        const unsubscribe = systemUnsubscribe.bind(null, subscription);
+        resolve(unsubscribe);
+    });
+}
+
+/**
+ * Remove a system subscription from all runtimes
+ */
+function systemUnsubscribe(subscription: any) {
+    subscription.unSubscriptions.forEach((runtime: any[]) => {
+        if (runtime.length) {
+            runtime.forEach((removeListener: any) => removeListener());
+        }
+    });
+    pendingRemoteSubscriptions.delete(subscription._id);
+}
+
+/**
+ * Subscribe to a system event in a remote runtime
+ */
+function applySystemSubscription(subscription: RemoteSubscription, runtime: PeerRuntime) {
```

Comment:
Can [this function](https://github.com/tgoc99/core/blob/03be9730f3b7ae18be562557808949fd55dd44a7/src/browser/remote_subscriptions.ts#L112) be reused here (a lot of code looks similar)

Reply:
I think its close.... I thought about trying to re-write this so that one fn could be used for both but after some advice decided against it.  Differences are: 
-  don't want cleanUpSubscription in the listener for system events
-  need to unsubscribe when runtime disconnects for system events (handled with combination of cleanUpSubscription & persistSubscription for other events)
- couple of other things but they're easy fixes

Let me know if you disagree.

Reply:
@tgoc99 Hard to tell for sure by taking a quick look, but if you thought of this and decided not to go that route, then okay. I'm fine with the fact that you considered it and still decided to put it in a separate function